File: src/cli-sample/Sample.cs
using System;

namespace Xt
{
    public class Sample
    {
        static readonly Action[] Samples =
        {
            PrintSimple.Main, PrintDetailed.Main, CaptureSimple.Main, RenderSimple.Main,
            CaptureAdvanced.Main, RenderAdvanced.Main, FullDuplex.Main, Aggregate.Main
         };

        static readonly string[] Names =
        {
            nameof(PrintSimple), nameof(PrintDetailed), nameof(CaptureSimple), nameof(RenderSimple),
            nameof(CaptureAdvanced), nameof(RenderAdvanced), nameof(FullDuplex), nameof(Aggregate)
        };

        static void RunSample(int index)
        {
            Console.WriteLine(Names[index] + ":");
            Samples[index]();
        }

        [STAThread]
        public static void Main(string[] args)
        {
            int index = args.Length == 1 ? int.Parse(args[0]) : -1;
            try
            {
                if (index >= 0)
                    RunSample(index);
                else
                    for (int i = 0; i < Samples.Length; i++)
                        RunSample(i);
            } catch (XtException e)
            { Console.WriteLine(XtAudio.GetErrorInfo(e.GetError())); }
        }
    }
}

File: src/java-sample/com/xtaudio/xt/sample/Sample.java
package com.xtaudio.xt.sample;

import com.xtaudio.xt.XtAudio;
import com.xtaudio.xt.XtException;

public class Sample {

    public static void main(String[] args) throws Exception {
        int index = -1;
        if (args.length == 1) {
            index = Integer.parseInt(args[0]);
        }
        try {
            if (index == -1 || index == 0) {
                System.out.println("PrintSimple:");
                PrintSimple.main(args);
            }
            if (index == -1 || index == 1) {
                System.out.println("PrintDetailed:");
                PrintDetailed.main(args);
            }
            if (index == -1 || index == 2) {
                System.out.println("CaptureSimple:");
                CaptureSimple.main(args);
            }
            if (index == -1 || index == 3) {
                System.out.println("RenderSimple:");
                RenderSimple.main(args);
            }
            if (index == -1 || index == 4) {
                System.out.println("CaptureAdvanced:");
                CaptureAdvanced.main(args);
            }
            if (index == -1 || index == 5) {
                System.out.println("RenderAdvanced:");
                RenderAdvanced.main(args);
            }
            if (index == -1 || index == 6) {
                System.out.println("FullDuplex:");
                FullDuplex.main(args);
            }
            if (index == -1 || index == 7) {
                System.out.println("Aggregate:");
                Aggregate.main(args);
            }
        } catch (XtException e) {
            System.out.println(XtAudio.getErrorInfo(e.getError()));
        }
    }
}